### PR TITLE
Added common components

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary
* Added Custom text input with `showPassword` prop
* Added Loadable HOC to show a spinner when the data is loading
* Added Custom button component
* Updated eslint dependencies

Know Issues:
* showing/hiding the text with secureTextEntry changes the cursor position:
https://github.com/facebook/react-native/issues/5859

#### Android
loadable android

#### iOS
loadable ios

#### Android
textinput android

#### iOS
textinput ios

### CODE REVIEW

**Comment by ignaciosantise** [comment]
[Include common components](https://trello.com/c/nl3dnJ9L/71-include-common-components)

File: generators/app/templates/src/app/components/CustomButton/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, { PureComponent } from 'react';
+import { TouchableOpacity, Image, ViewPropTypes } from 'react-native';
+import PropTypes from 'prop-types';
+
+import CustomText from '../CustomText';
+
+import styles from './styles';
+
+export default class CustomButton extends PureComponent {
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomButton/index.js]: Keep `export default` at the bottom of this file

File: generators/app/templates/src/app/components/CustomButton/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  static VARIANTS = ['borderless', 'radial', 'black', 'green', 'small', 'white', 'gray'];
+
+  customStyles = () =>
+    CustomButton.VARIANTS.map(variant => (this.props[variant] ? styles[variant] : null)).filter(
+      style => style !== null
+    );
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomButton/index.js]: I'd extract this function and reuse it in `customTextStyles`

File: generators/app/templates/src/app/components/CustomButton/index.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  activeOpacity: 1
+};
+
+CustomButton.propTypes = {
+  activeOpacity: PropTypes.number,
+  disabled: PropTypes.bool,
+  icon: PropTypes.number,
+  iconStyle: ViewPropTypes.style, // eslint-disable-line react/no-typos
+  onPress: PropTypes.func.isRequired,
+  style: ViewPropTypes.style, // eslint-disable-line react/no-typos
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomButton/index.js]: If applies, I'd delete this rule. What do you think ? @WFolini @sbalay @IgnacioAbadie @ignaciosantise

File: generators/app/templates/src/app/components/CustomButton/styles.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import { transparent, black, green, gray } from '../../../constants/colors';
+
+const iconSize = 20;
+
+const borderlessStyle = {
+  borderWidth: 0,
+  backgroundColor: transparent
+};
+
+export default StyleSheet.create({
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomButton/styles.js]: `export default` at the bottom

File: generators/app/templates/src/app/components/CustomText/index.js [hunk 5]
```diff
@@ -25,9 +25,9 @@ export default class CustomText extends PureComponent {
   ];
 
   customStyles = () =>
-    CustomText.VARIANTS
-      .map(variant => (this.props[variant] ? styles[variant] : null))
-      .filter(style => style !== null);
+    CustomText.VARIANTS.map(variant => (this.props[variant] ? styles[variant] : null)).filter(
+      style => style !== null
+    );
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomText/index.js]: I'd reuse the function extracted from `CustomButton`

File: generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/index.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { TouchableOpacity, Image } from 'react-native';
+
+import icEyePng from '../../assets/ic_eye.png';
+import icNoeyePng from '../../assets/ic_noeye.png';
+
+import styles from './styles';
+
+const ShowPassword = ({ onShowPassword, passwordVisible }) => (
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/index.js]: `function` or `arrow function` 🤔  To be or not to be

File: generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/styles.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { StyleSheet } from 'react-native';
+
+import { blue } from '../../../../../constants/colors';
+
+export default StyleSheet.create({
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/styles.js]: `export default`

File: generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/styles.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  container: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    bottom: -4
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/styles.js]: Mmm, why this ? This doesn't have issues with `Android` overflow ?
If so, I'd put `marginBottom = -4`

File: generators/app/templates/src/app/components/CustomTextInput/index.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+  handleShowPassword = () => {
+    this.setState(prevState => ({ showPassword: !prevState.showPassword }));
+  };
+
+  render() {
+    const placeholderTextColor = this.props.value ? transparent : this.props.placeholderTextColor;
+
+    return (
+      <View>
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomTextInput/index.js]: `<Fragment>` ?

File: generators/app/templates/src/app/components/CustomTextInput/index.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        {this.props.title && (
+          <CustomText darkGray style={[styles.title, this.props.titleStyles]}>
+            {this.props.title}
+          </CustomText>
+        )}
+        <View
+          style={[
+            this.props.multiline ? styles.multilineContainer : styles.container,
+            this.props.bottomBorder ? styles.bottomBorder : {},
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomTextInput/index.js]:
```
this.props.bottomBorder && styles.bottomBorder
```
?

File: generators/app/templates/src/app/components/CustomTextInput/index.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  clearButtonMode: PropTypes.oneOf('never', 'while-editing', 'unless-editing', 'always'),
+  returnKeyType: PropTypes.oneOf('done', 'go', 'next', 'search'),
+  keyboardType: PropTypes.oneOf('default', 'numeric', 'email-address', 'phone-pad'),
+  underlineColorAndroid: PropTypes.string,
+  maxHeight: PropTypes.number,
+  autoCorrect: PropTypes.bool,
+  multiline: PropTypes.bool,
+  placeholder: PropTypes.string,
+  titleStyles: Text.propTypes.style, // eslint-disable-line react/no-typos
+  textStyles: Text.propTypes.style, // eslint-disable-line react/no-typos
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomTextInput/index.js]: Same comment here.
Mmm isn't this deprecated ? 🤔  I've searched for `TextPropTypes` but I've found nothing.
For example, `View.propTypes.style` is deprecated.

File: generators/app/templates/src/app/components/CustomTextInput/styles.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
+import { StyleSheet } from 'react-native';
+
+import { gray, black, transparent } from '../../../constants/colors';
+
+export default StyleSheet.create({
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomTextInput/styles.js]: `export default`

File: generators/app/templates/src/app/components/Loadable/styles.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import { StyleSheet } from 'react-native';
+
+import { white } from '../../../constants/colors';
+
+export default StyleSheet.create({
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/Loadable/styles.js]: `export default`

File: generators/app/templates/src/app/components/CustomButton/index.js (hunk 3 above)
Comment (IgnacioAbadie) [reply generators/app/templates/src/app/components/CustomButton/index.js]: 👍

File: generators/app/templates/src/app/components/CustomButton/styles.js (hunk 4 above)
Comment (IgnacioAbadie) [reply generators/app/templates/src/app/components/CustomButton/styles.js]: i think that we have all export default of styles file like @ignaciosantise wrote.
In components maybe, but in style files i like this. Really there is not difference

File: generators/app/templates/src/app/components/CustomText/index.js (hunk 5 above)
Comment (IgnacioAbadie) [reply generators/app/templates/src/app/components/CustomText/index.js]: 👍

File: generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/index.js (hunk 6 above)
Comment (IgnacioAbadie) [reply generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/index.js]: jaja we have to define one of

**Comment by IgnacioAbadie** [comment]
@ignaciosantise check the conflics

File: generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/index.js (hunk 6 above)
Comment (ignaciosantise) [reply generators/app/templates/src/app/components/CustomTextInput/components/ShowPassword/index.js]: arrow function 👼

File: generators/app/templates/src/app/components/CustomTextInput/index.js (hunk 10 above)
Comment (ignaciosantise) [reply generators/app/templates/src/app/components/CustomTextInput/index.js]: Yeeep!

File: generators/app/templates/src/app/components/CustomTextInput/index.js (hunk 11 above)
Comment (ignaciosantise) [reply generators/app/templates/src/app/components/CustomTextInput/index.js]: View.propTypes.style is deprecated, but there's no info about Text.propTypes. and it doesnt throw a warning or sth

File: generators/app/templates/src/app/components/CustomButton/styles.js (hunk 4 above)
Comment (ignaciosantise) [reply generators/app/templates/src/app/components/CustomButton/styles.js]: if we are changing the export position of Stylesheets, we should change all of them in another PR

**Comment by ignaciosantise** [comment]
@WFolini great! Assets changed ;)

About the props, think that configuration will depend on the project. Not all projects follow that design line. I think that colors and sizes should be customizable by props

File: generators/app/templates/src/app/components/CustomButton/styles.js (hunk 4 above)
Comment (mvbattan) [reply generators/app/templates/src/app/components/CustomButton/styles.js]: Agree, let's do it in another PR

File: generators/app/files.js [hunk 14]
```diff
@@ -43,6 +43,44 @@ module.exports.CUSTOM_TEXT_PATH = `${module.exports.COMPONENTS_PATH}/CustomText`
...
+module.exports.CUSTOM_TEXT_INPUT_ASSETS_PATH = `${module.exports.CUSTOM_TEXT_INPUT_PATH}/assets`;
+module.exports.CUSTOM_TEXT_INPUT_ASSETS_VISIBILITY = `${
+  module.exports.CUSTOM_TEXT_INPUT_ASSETS_PATH
+}/ic_visibility.png`;
+module.exports.CUSTOM_TEXT_INPUT_ASSETS_VISIBILITY_2X = `${
+  module.exports.CUSTOM_TEXT_INPUT_ASSETS_PATH
+}/ic_visibility@2x.png`;
+module.exports.CUSTOM_TEXT_INPUT_ASSETS_VISIBILITY_3X = `${
+  module.exports.CUSTOM_TEXT_INPUT_ASSETS_PATH
+}/ic_visibility@3x.png`;
```
Comment (mvbattan) [inline generators/app/files.js]: Dunno. It's likely this `ic_visibility` is being repeated over and over

**Pull request merged**